universal-application-tool-0.0.1/app/models/Application.java

```diff
@@ -46,6 +45,7 @@ public Application(Applicant applicant, Program program, LifecycleStage lifecycl
     setApplicantData(applicant.getApplicantData());
     this.program = program;
     this.lifecycleStage = lifecycleStage;
+    setSubmitTimeToNow();
```

Comment:
issue (blocking): this will run every time a new `Application` object is initialized, regardless of whether it represents an existing application already persisted in the database or a newly created/submitted one

Reply:
Updated.